### ISSUE TITLE
Calling wiser.boost_heating results in 'override' rather than 'boost'.

### ISSUE THREAD

**Issue by charlesomer** [asantaga/wiserHomeAssistantPlatform#242]
Hi,

I've just noticed that when I call the `wiser.boost_heating` service (from the developer tools for example) it does set the temperature correctly but it uses 'override' instead of 'boost' so it won't reset after an hour for example.

For reference I've tried this in `normal`, `boost` and `boost only on auto` modes

Thank you
Charles

**Comment by msp1974** [comment]
@charlesomer can you provide more details.  Ie what parameters did you pass to the service call?  Does it work with any parameters or always shows override?  Does the wiser app show as it is overridden or boosted? Version of integration in use?

**Comment by msp1974** [comment]
Could you also please follow this to provide the hub json info after calling the service and it showing as override.  https://github.com/asantaga/wiserHomeAssistantPlatform#providing-your-hub-output-as-json-files
Thx

**Comment by charlesomer** [comment]
Service call:
```
service: wiser.boost_heating
data:
  temperature: 21
target:
  entity_id: climate.wiser_hallway
```
It always seems to say 'override' unless it's already in the 'boost' mode, in which case it will change the temperature to whatever is set in the service call but not reset the boost duration.

The wiser app shows it as overridden.

v3.0.22
[wiser_data.zip](https://github.com/asantaga/wiserHomeAssistantPlatform/files/8205853/wiser_data.zip)

Thank you :)

**Comment by msp1974** [comment]
Hmm.. ok will have a look at this tonight.  Presume if you use temp_delta instead of temperature it does boost?  Setting temperature is not actually boost but an override for duration.  However, seeing your json output it does not seem to have set the duration.  I wonder if this changed in a version of the hub firmware.

**Comment by charlesomer** [comment]
Hmm, interestingly, now when I use the development tools section of HA to call the service, it does indeed use Boost as expected. However, if I create a button in a custom Dashboard which calls the same service with the same options, it uses Override :/ Very odd!

**Comment by msp1974** [comment]
> Service call:
> 
> ```
> service: wiser.boost_heating
> data:
>   temperature: 21
> target:
>   entity_id: climate.wiser_hallway
> ```
> 
> It always seems to say 'override' unless it's already in the 'boost' mode, in which case it will change the temperature to whatever is set in the service call but not reset the boost duration.
> 
> The wiser app shows it as overridden.
> 
> v3.0.22 [wiser_data.zip](https://github.com/asantaga/wiserHomeAssistantPlatform/files/8205853/wiser_data.zip)
> 
> Thank you :)

I think I see the issue.  You are not setting a duration / it is not using the default if not set therefore it is just overriding the temp for no time period.

There is a weird behaviour in the dev tools where a default value is set in that it does not use it unless you change the default value.  Ie time is default 60, if you change to 59 then back to 60 it uses it.  Equally if you are trying to do this with a button, your yaml should be:

```
service: wiser.boost_heating
data:
  temperature: 21
  time_period: 60
target:
  entity_id: climate.wiser_hallway
```

**Comment by msp1974** [comment]
Will provide fix to use default boost time if not set in service call in next release.  For now add the time_period.

**Comment by charlesomer** [comment]
Thank you! That was exactly it 👍 It’s working as expected now 😊

**Comment by msp1974** [comment]
Fixed in v3.0.23
